File: PrivateVault.ai-main/sdk/node/src/retries.js
export class RetryConfig {
  constructor({ attempts = 3, backoffSeconds = 0.25, retryStatuses = [429, 500, 502, 503, 504] } = {}) {
    this.attempts = attempts;
    this.backoffSeconds = backoffSeconds;
    this.retryStatuses = retryStatuses;
  }
}

export async function backoffSleep(attempt, baseSeconds) {
  const delay = baseSeconds * Math.pow(2, attempt - 1);
  await new Promise((resolve) => setTimeout(resolve, delay * 1000));
}

File: PrivateVault.ai-main/sdk/python/privatevault_sdk/retries.py
import time
from dataclasses import dataclass
from typing import Iterable


@dataclass
class RetryConfig:
    attempts: int = 3
    backoff_seconds: float = 0.25
    retry_statuses: Iterable[int] = (429, 500, 502, 503, 504)


def backoff_sleep(attempt: int, base: float) -> None:
    time.sleep(base * (2 ** (attempt - 1)))
